src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -2322,7 +2322,8 @@ UNATIVE_OFFSET emitter::emitInsSizeAM(instrDesc* id, code_t code)
         }
         else
         {
-            if (dspIsZero && baseRegisterRequiresDisplacement(reg) && !baseRegisterRequiresDisplacement(rgx))
+            if (dspIsZero && baseRegisterRequiresDisplacement(reg) && !baseRegisterRequiresDisplacement(rgx) &&
+                !isFloatReg(rgx))
```

Comment:
This was failing for a gather instruction, which uses the VSIB format, but for which we can't reverse `reg` and `rgx` since the latter is a vector instruction.

Reply:
I would find the change confusing without this explanation, should it be added as a comment?